### PR TITLE
feat(plugin-calendar): support for configuring the background color of calendar events

### PR DESCRIPTION
**Pull request by WangMin66**
### This is a ...
- [x] New feature
- [ ] Improvement
- [ ] Bug fix
- [ ] Others

### Showcase

20240914192017_rec_
### Changelog
| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Support for configuring the background color of calendar events |
| 🇨🇳 Chinese | 支持配置日历事件的背景颜色 |
### Docs
| Language   | Link |
| ---------- | --------- |
| 🇺🇸 English |  [Title](https://github.com/nocobase/docs/pull/161)    |
| 🇨🇳 Chinese |  [合并之后更改](https://github.com/nocobase/docs/pull/161)  |
### Checklists
- [x] All changes have been self-tested and work as expected
- [x] Test cases are updated/provided or not needed
- [x] Doc is updated/provided or not needed
- [x] Component demo is updated/provided or not needed
- [x] Changelog is provided or not needed
- [x] Request a code review if it is necessary

### CODE REVIEW

**Comment by zhangzhonghe** [comment]
@WangMin66 首先，建议按照我们的 PR 模板，把需要写的信息补一下。感谢。

**Comment by zhangzhonghe** [comment]
@WangMin66 另外，现在 build 是报错的，需要先修复下。

File: packages/plugins/@nocobase/plugin-calendar/src/client/calendar/Calendar.tsx [hunk 1]
```diff
@@ -10,31 +10,32 @@
...
-import { CalendarRecordViewer, findEventSchema } from './CalendarRecordViewer';
 import Header from './components/Header';
 import { CalendarToolbarContext } from './context';
 import GlobalStyle from './global.style';
 import { useCalenderHeight } from './hook';
 import useStyle from './style';
 import type { ToolbarProps } from './types';
 import { formatDate } from './utils';
+import style from './style';
+import { color } from 'packages/plugins/@nocobase/plugin-mock-collections/src/server/field-interfaces';
```
Comment (zhangzhonghe) [inline packages/plugins/@nocobase/plugin-calendar/src/client/calendar/Calendar.tsx:38]: 现在不支持引用绝对路径，需要是相对路径。或者 `import * from '@nocobase/plugin-mock-collections'` 这样引用。

**Comment by WangMin66** [comment]
### This is a ...
- [ ] New feature
- [x] Improvement
- [ ] Bug fix
- [ ] Others
### Motivation

### Description 

### Related issues
### Showcase

### Changelog
| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | The calendar can select 1 enumeration field and directly take the color setting of the enumeration field The background color of the event title displayed in the calendar is set directly with the color of the enumeration field |
| 🇨🇳 Chinese | 日历可以选择1个 枚举字段， 直接取枚举字段的颜色设置,日历中显示的事件标题的背景颜色，直接用枚举字段的颜色设置 |
### Docs
| Language   | Link |
| ---------- | --------- |
| 🇺🇸 English |      |
| 🇨🇳 Chinese |    |
### Checklists
- [ ] All changes have been self-tested and work as expected
- [ ] Test cases are updated/provided or not needed
- [ ] Doc is updated/provided or not needed
- [ ] Component demo is updated/provided or not needed
- [ ] Changelog is provided or not needed
- [ ] Request a code review if it is necessary

File: packages/plugins/@nocobase/plugin-calendar/src/client/calendar/Calendar.tsx [hunk 2]
```diff
@@ -216,60 +262,78 @@ export const Calendar: any = withDynamicSchemaProps(
...
-                console.log('onDoubleClickEvent');
-              }}
-              onSelectEvent={(event) => {
-                const record = dataSource?.find((item) => item[fieldNames.id] === event.id);
-                if (!record) {
-                  return;
-                }
-                record.__event = { ...event, start: formatDate(dayjs(event.start)), end: formatDate(dayjs(event.end)) };
+          <GlobalStyle />
+          <CalendarRecordViewer visible={visible} setVisible={setVisible} record={record} />
```
Comment (zhangzhonghe) [inline packages/plugins/@nocobase/plugin-calendar/src/client/calendar/Calendar.tsx:293]: 这里为什么要这样改？是遇到什么问题了吗？现在这样改，导致一些功能失效了。

File: packages/plugins/@nocobase/plugin-calendar/src/client/calendar/Calendar.tsx [hunk 3]
```diff
@@ -157,29 +161,71 @@ const useEvents = (dataSource: any, fieldNames: any, date: Date, view: (typeof W
...
+      </DeleteEventContext.Provider>
+    )
+  );
 };
 
 export const Calendar: any = withDynamicSchemaProps(
   observer(
     (props: any) => {
-      const [visible, setVisible] = useState(false);
-      const { openPopup } = usePopupUtils({
```
Comment (zhangzhonghe) [inline packages/plugins/@nocobase/plugin-calendar/src/client/calendar/Calendar.tsx:168]: 这个是不能去掉的。弹窗是需要通过 `openPopup` 这个方法打开的。

**Comment by zhangzhonghe** [comment]
Completed in https://github.com/nocobase/nocobase/pull/5290

**Pull request closed without merging**